ZitiUpdateService/UpdateService.cs

```diff
@@ -89,19 +90,33 @@ public partial class UpdateService : ServiceBase {
 			svr.DoUpdateCheck = DoUpdateCheck;
 			svr.TriggerUpdate = TriggerUpdate;
 			svr.SetAutomaticUpdateDisabled = SetAutomaticUpdateDisabled;
+			svr.SetAutomaticUpdateURL = SetAutomaticUpdateURL;
 
 			string assemblyVersionStr = Assembly.GetExecutingAssembly().GetName().Version.ToString(); //fetch from ziti?
 			assemblyVersion = new Version(assemblyVersionStr);
 			asmDir = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
 			updateFolder = Path.Combine(asmDir, "updates");
-			if (!Directory.Exists(updateFolder))
-			{
+			if (!Directory.Exists(updateFolder)) {
 				Directory.CreateDirectory(updateFolder);
 			}
 		}
 
-		private void CurrentSettings_OnConfigurationChange(object sender, ControllerEvent e)
-		{
+		private SvcResponse SetAutomaticUpdateURL(string url) {
+			SvcResponse r = new SvcResponse();
+			if (url == null || !url.StartsWith("http")) {
+				r.Code = (int)ErrorCodes.URL_INVALID;
+				r.Error = $"The url supplied is invalid: \n{url}\n";
+				r.Message = "Failure";
+				return r;
+			} else {
+				CurrentSettings.AutomaticUpdateURL = url;

```

Comment:
add sanity check to make sure url is avaialble and response is in the correct format